src/sentry/models/group.py

```diff
@@ -799,7 +799,7 @@ def get_event_type(self):
         """
         return self.data.get("type", "default")
 
-    def get_event_metadata(self) -> Mapping[str, str]:
+    def get_event_metadata(self) -> Mapping[str, Any]:
```

Comment:
metadata is anything JSON-able. Since this has to work in Python 3.8 we cannot define recursive type, so I'm using `Any` as a workaround.